src/Storages/System/StorageSystemDistributionQueue.cpp

```diff
@@ -0,0 +1,116 @@
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Storages/System/StorageSystemDistributionQueue.h>
+#include <Storages/Distributed/DirectoryMonitor.h>
+#include <Storages/StorageDistributed.h>
+#include <Storages/VirtualColumnUtils.h>
+#include <Access/ContextAccess.h>
+#include <Common/typeid_cast.h>
+#include <Databases/IDatabase.h>
+
+
+namespace DB
+{
+
+
+NamesAndTypesList StorageSystemDistributionQueue::getNamesAndTypes()
+{
+    return {
+        { "database",              std::make_shared<DataTypeString>() },
+        { "table",                 std::make_shared<DataTypeString>() },
+        { "data_path",             std::make_shared<DataTypeString>() },
+        { "is_blocked",            std::make_shared<DataTypeUInt8>()  },
+        { "error_count",           std::make_shared<DataTypeUInt64>() },
+        { "data_files",            std::make_shared<DataTypeUInt64>() },
+        { "data_compressed_bytes", std::make_shared<DataTypeUInt64>() },
+        { "last_exception",        std::make_shared<DataTypeString>() },
+    };
+}
+
+
+void StorageSystemDistributionQueue::fillData(MutableColumns & res_columns, const Context & context, const SelectQueryInfo & query_info) const
+{
+    const auto access = context.getAccess();
+    const bool check_access_for_databases = !access->isGranted(AccessType::SHOW_TABLES);
+
+    std::map<String, std::map<String, StoragePtr>> tables;
+    for (const auto & db : DatabaseCatalog::instance().getDatabases())
+    {
+        /// Lazy database can not contain distributed tables
+        if (db.second->getEngineName() == "Lazy")
+            continue;
+
+        const bool check_access_for_tables = check_access_for_databases && !access->isGranted(AccessType::SHOW_TABLES, db.first);
+
+        for (auto iterator = db.second->getTablesIterator(context); iterator->isValid(); iterator->next())
+        {
+            if (!dynamic_cast<const StorageDistributed *>(iterator->table().get()))
```

Comment:
And here we have terrible bug to to deficiency of the iterator interface:
https://github.com/ClickHouse/ClickHouse/pull/11348